components/SearchSandbox/utils/dataField.js

```diff
@@ -18,7 +18,11 @@ const getSubFields = (mappings, field, types) => [
 // based on the selected-field(s)
 const generateDataField = (component, selectedFields, mappings) => {
 	// no need to get subfields if it is a result component
-	if (component === 'ReactiveList') return selectedFields;
+	if (component === 'ReactiveList') {
+		const dataField = mappings[selectedFields].fields.length > 0 ?
+			`${selectedFields}.${mappings[selectedFields].originalFields.keyword.type}` : selectedFields;
```

Comment:
Can you please explain what's happening here?

I don't think this logic is accurate. It is not always the case that types with sub-fields will always have `keyword` included.